app/services/declarations/create.rb

```diff
@@ -165,7 +166,25 @@ def validate_has_passed?
     end
 
     def valid_course_identifier_for_participant_outcome?
-      CourseGroup.joins(:courses).where(name: %w[leadership specialist]).map(&:courses).flatten.uniq.map(&:identifier).include?(course_identifier)
+      CourseGroup.joins(:courses).leadership_or_specialist.where(courses: { identifier: course_identifier }).exists?
+    end
+
+    def create_participant_outcome!
+      return unless validate_has_passed?
+
+      service = ParticipantOutcomes::Create.new(
+        lead_provider:,
+        participant:,
+        course_identifier:,
+        state: has_passed.to_s == "true" ? "passed" : "failed",
+        completion_date: declaration_date.rfc3339,
+      )
+
+      if service.valid?
+        service.create_outcome
+      else
+        raise ArgumentError, I18n.t(:cannot_create_completed_declaration)
```

Comment:
in ECF we have been using this pattern however in NPQ we were hoping to move towards this failing at the initial `valid?` stage of the create declaration service rather than fail from within the transaction, is it possible to instantiate and check this service is valid when we do all other validations instead?

Reply:
Hi @cwrw I tried that idea but the outcomes create service class relies on the completed declaration to exists as part of its own validation. If we move this code to be a validation in the declarations create class, the completed declaration won't yet be present and the outcomes service class will be always invalid.